### PR TITLE
Make topoclustering work with new theta-module merged readout 

### PR DESCRIPTION
**Pull request by giovannimarchiori**
This PR and related ones for FCCDetectors and LAr_scripts make topoclustering work with the new theta-module readout for the FCC-ee calorimeter.
I would like to initiate the review to discuss the code structure - I will then remove commented code and debug messages later. 

The PR also contain a couple of fixes to what I think are bugs in the current code, and removes warning in FCCSWGridModuleThetaMerged.h from a int vs uint comparison

Related PRs: 
https://github.com/HEP-FCC/k4RecCalorimeter/pull/51
https://github.com/BrieucF/LAr_scripts/pull/13

Tagging @gartrog @BrieucF @dasphy

### CODE REVIEW

**Comment by giovannimarchiori** [comment]
Hi @BrieucF I think this PR and the related ones are now good for review. I have added the treatment of the diagonal neighbours, cleaned up the code, removed debugging statements, updated the detector parameters to get back 1536 modules. main thing missing is barrel ECAL-HCAL matching (to find neighbours across different volumes) but this can be tackled by a separate PR once we will start looking at hadronic clusters.

File: Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ class FCCSWGridModuleThetaMerged : public GridTheta {
    *   return The number of merged cells in theta
    */
   inline int mergedThetaCells(const int layer) const {
-    if (layer<m_mergedCellsTheta.size())
+    if (layer < (int) m_mergedCellsTheta.size())
```
Comment (kjvbrt) [inline Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: Are you sure that `m_mergedCellsTheta.size()` will not be larger than int numeric limit?

Comment (giovannimarchiori) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: yes, cells are grouped in small numbers (typically between 1 and 4)

Comment (kjvbrt) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: OK, can you add this comment to the code?

Comment (giovannimarchiori) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: sorry answer was not correct - you asked about size() of the vector, not about the content of the vector. The size of the vector is equal to the number of layers of the calo, which is 12 in the baseline design and anyway O(10)

Comment (giovannimarchiori) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: now that I clarified that the size of the vector is equal to nLayers (as written also when the vector is declared, and do you think I should really add this comment?

Comment (kjvbrt) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: Yes, to have the justification for the casting to `int` directly in the code.

Comment (kjvbrt) [reply Detector/DetSegmentation/include/DetSegmentation/FCCSWGridModuleThetaMerged.h:66]: Nice

**Pull request merged**